### PR TITLE
iba dashboard review

### PR DESCRIPTION
**Pull request by chrismarget-j**
It looks to me like we only fetch the dashboard in `Create()` and `Update()` so that we can populate the `UpdatedBy` attribute.

Is that right?

It doesn't seem like a very important/interesting attribute to me. I'm inclined to drop terraform support for that attribute (it's not in the GUI, I think?) so we can skip the extra fetch step.

### CODE REVIEW

File: apstra/resource_iba_dashboard.go [hunk 1]
```diff
@@ -39,46 +39,40 @@ func (o *resourceIbaDashboard) Create(ctx context.Context, req resource.CreateRe
...
-	// 		fmt.Sprintf("error locking blueprint %q mutex", plan.BlueprintId.ValueString()),
-	// 		err.Error())
-	// 	return
-	// }
+	// Convert the plan into an API Request
+	dashReq := plan.Request(ctx, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
```
Comment (chrismarget-j) [inline apstra/resource_iba_dashboard.go:55]: - Moved the request creation so it's closer to where it's used
- Check for errors
- Delete the dead blueprint lock code (dashboards aren't part of the blueprint for "commit" reasons, I guess?)

File: apstra/resource_iba_dashboard.go [hunk 2]
```diff
@@ -39,46 +39,40 @@ func (o *resourceIbaDashboard) Create(ctx context.Context, req resource.CreateRe
...
+	plan.LoadApiData(ctx, api, &resp.Diagnostics)
 	if resp.Diagnostics.HasError() {
 		return
 	}
-	newstate.BlueprintId = plan.BlueprintId
+
 	// Set state
-	//
-	resp.Diagnostics.Append(resp.State.Set(ctx, &newstate)...)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
```
Comment (chrismarget-j) [inline apstra/resource_iba_dashboard.go:75]: I don't think we need a `newState` object here, it should be safe/reasonable to invoke `LoadApiData` on the `plan` (and elsewhere `state`)

File: apstra/resource_iba_dashboard.go [hunk 3]
```diff
@@ -88,35 +82,34 @@ func (o *resourceIbaDashboard) Read(ctx context.Context, req resource.ReadReques
...
 
-	var err error
-	var api *apstra.IbaDashboard
-
 	bpClient, err := o.client.NewTwoStageL3ClosClient(ctx, apstra.ObjectId(state.BlueprintId.ValueString()))
 	if err != nil {
+		if utils.IsApstra404(err) {
+			resp.State.RemoveResource(ctx)
+			return
+		}
```
Comment (chrismarget-j) [inline apstra/resource_iba_dashboard.go:90]: Dashboard is gone if blueprint is gone.

**Review by rajagopalans — APPROVED** [review]

File: apstra/resource_iba_dashboard.go (hunk 3 above)
Comment (rajagopalans) [reply apstra/resource_iba_dashboard.go:90]: So, I need to remove the resource right?

Comment (rajagopalans) [reply apstra/resource_iba_dashboard.go:90]: So, I do need to remove the resource right?

**Comment by rajagopalans** [comment]
> It looks to me like we only fetch the dashboard in `Create()` and `Update()` so that we can populate the `UpdatedBy` attribute.
> 
> Is that right?
There is also the "Default" and "Predefined Dashboards" field.
It might help if we want to do a future filter like 'give me all dashboards created from this predefined dashboard' or 'give me all default dashboards'.
I don't have a strong opinion  on dropping or keeping support for these fields.

**Pull request merged**